resource_usage_recorder/system_recorder/battery/abstract_battery_usage_recorder.py

```
@@ -1,16 +1,16 @@
 from abc import ABC, abstractmethod
 from typing import TextIO, Optional
 
-import pandas as pd
+from resource_usage_recorder import MetricResult, MetricRecorder
 
 
-class AbstractBatteryMonitor(ABC):
+class AbstractBatteryUsageRecorder(MetricRecorder, ABC):
     @abstractmethod
     def check_if_battery_plugged(self):
         pass
 
     @abstractmethod
-    def get_current_metrics(self, battery_df: pd.DataFrame, time_interval: float):
+    def get_current_metrics(self) -> MetricResult:
```

since both children return SystemBatteryResults, return it.
In the null battery child - think about a different implementation that will match the type hint. Maybe raise an error instead of doing pass.